geode-redis/src/integrationTest/java/org/apache/geode/redis/general/ExistsIntegrationTest.java

```diff
@@ -0,0 +1,348 @@
+  }
+
+  @Test
+  public void shouldReturn0_givenListDoesNotExist() {
+    String listKey = "listKey";
+    String listValue = "listValue";
+
+    jedis.lpush(listKey, listValue);
+    jedis.del(listKey);
+
+    assertThat(jedis.exists(toArray(listKey))).isEqualTo(0L);
+  }
+
+  @Test
+  public void shouldReturn1_givenBitMapExists() {
+    String bitMapKey = "bitMapKey";
+    long offset = 1L;
+    String bitMapValue = "0";
+
+    jedis.setbit(bitMapKey, offset, bitMapValue);
+
+    assertThat(jedis.exists(toArray(bitMapKey))).isEqualTo(1L);
+  }
+
+  @Test
+  public void shouldReturn0_givenBitMapDoesNotExist() {
+    String bitMapKey = "bitMapKey";
+    long offset = 1L;
+    String bitMapValue = "0";
+
+    jedis.setbit(bitMapKey, offset, bitMapValue);
+    jedis.del(bitMapKey);
+
+    assertThat(jedis.exists(toArray(bitMapKey))).isEqualTo(0L);
+  }
+
+  @Test
+  public void shouldReturnTotalNumber_givenMultipleKeys() {
+    String key1 = "key1";
+    String key2 = "key2";
+
+    jedis.set(key1, "value1");
+    jedis.set(key2, "value2");
+
+    assertThat(jedis.exists(toArray(key1, "doesNotExist1", key2, "doesNotExist2"))).isEqualTo(2L);
+  }
+
+  @Test
+  public void shouldCorrectlyVerifyKeysExistConcurrently() throws InterruptedException {
+    int iterationCount = 5000;
+    setKeys(jedis, iterationCount);
+
+    AtomicLong existsCount = new AtomicLong(0);
+
+    Thread thread1 =
+        new LoopingThread((i) -> existsCount.addAndGet(jedis.exists(toArray("key" + i))),
+            iterationCount);
+    Thread thread2 =
+        new LoopingThread((i) -> existsCount.addAndGet(jedis2.exists(toArray("key" + i))),
+            iterationCount);
+
+    thread1.start();
+    thread2.start();
+    thread1.join();
+    thread2.join();
+
+    assertThat(existsCount.get()).isEqualTo(2 * iterationCount);
+  }
+
+  @Test
+  public void shouldCorrectlyVerifyKeyExistsConcurrentlyWhileDeletingAndSettingKey()
+      throws InterruptedException {
+    int iterationCount = 5000;
+
+    Thread loopingThread1 = new LoopingThread((i) -> jedis.set("key", "value"), iterationCount);
+    Thread loopingThread2 = new LoopingThread((i) -> jedis2.exists(toArray("key")), iterationCount);
+    Thread loopingThread3 = new LoopingThread((i) -> jedis3.del("key"), iterationCount);
+
+    loopingThread1.start();
+    loopingThread2.start();
+    loopingThread3.start();
+    loopingThread1.join();
+    loopingThread2.join();
+    loopingThread3.join();
+  }
+
+  public String[] toArray(String... strings) {
+    return strings;
+  }
+
+  private void setKeys(Jedis jedis, int iterationCount) {
+    for (int i = 0; i < iterationCount; i++) {
+      jedis.set("key" + i, "value" + i);
+    }
+  }
+
+  private void existsKeys(AtomicLong existsCount, Jedis jedis, int iterationCount) {
+    for (int i = 0; i < iterationCount; i++) {
+      existsCount.addAndGet(jedis.exists("key" + i, "doesNotExist" + i));
+    }
+  }
+
+  private class LoopingThread extends Thread {
```

Comment:
I don't think it's absolutely critical for this commit, but we probably need to standardize on an approach for (a) spawning and executing threads, and (b) iterated runnables. There's a few different approaches scattered across the tests.

Reply:
I totally agree!  Let's have a techtro!